solver/src/driver.rs

```diff
@@ -312,19 +287,21 @@ impl Driver {
                 .expect("Price of native token must be known.");
 
         futures::stream::iter(settlements)
-            .filter_map(|settlement| async {
-                let surplus = settlement.settlement.total_surplus(prices);
+            .filter_map(|settlement_with_solver_id| async {
+                let (solver_id, settlement) = settlement_with_solver_id;
```

Comment:
any reason we don't just directly declare the tuple here?

Reply:
Borrow checker refused to compile it for some reason. I suspect it may be a bug.